app/entry.server.tsx

```diff
@@ -149,20 +156,37 @@ function handleBotRequest(
   });
 }
 
-function handleBrowserRequest(
+async function handleBrowserRequest(
   request: Request,
   responseStatusCode: number,
   responseHeaders: Headers,
   remixContext: EntryContext
 ) {
+  let instance = createInstance();
+  let ns = i18next.getRouteNamespaces(remixContext);
+  let lng = getLng(request);
+  await instance
+    .use(initReactI18next) // Tell our instance to use react-i18next
+    .use(LanguageDetector)
+    .init({
+      ...i18n, // spread the configuration
+      ns, // The namespaces the routes about to render wants to use
+      lng, // The language to use
+      resources: { en: { common: en }, fr: { common: fr } },
```

Comment:
If we store the `supportedLngs` in shelf.config we can then have a function that generates this object as we need to use it in a couple of different places. Adding a new language should be as simple as adding it to supportedLngs in shelf.config and creating the empty translation file.